src/payment/data/sagas.js

```diff
@@ -22,10 +22,14 @@ export function* handleFetchBasket() {
   try {
     const result = yield call(PaymentApiService.getBasket);
     yield put(fetchBasketSuccess(result));
-    if (result.voucher === undefined) {
+    if (result.coupons[0] === undefined) {
```

Comment:
Note that this check differs from above in the coupon sagas.  Up there we check to see if coupons is undefined - here we're assuming it's an array.  Only one of those should be true/they should be consistent... (this is another "How is the backend implemented?" question, but the answer should probably (I hope!) reveal one of these two sagas to have a bug.

Reply:
Changed to `result.coupons.length === 0`, when coupon is removed, the basket for coupons is an empty array.